src/components/ui/Button.jsx

```diff
@@ -4,13 +4,19 @@ import PropTypes from 'prop-types';
 
 const Button = ({
   children, icon, variant = 'secondary', type = 'button',
-  href, onClick, className, ...rest
+  disabled, href, onClick, className, ...rest
 }) => {
   const Tag = href ? 'a' : 'button';
 
   return <Tag
     type={type}
-    className={classnames('button', { [`button--${variant}`]: variant }, className)}
+    disabled={disabled}
+    className={classnames(
+      'button',
+      { [`button--${variant}`]: variant },
+      { ['button--disabled']: disabled },
```

Comment:
We don't need a variant for that. You can target with the `:disabled` CSS selector.